browser/src/react-redux/App/SignupPage.jsx

```diff
@@ -0,0 +1,159 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import setUser from './setUser.action'
+import axios from 'axios';
+
+const initialState = {
+  confirmPassword: '',
+  email: '',
+  password: '',
+  emailError: '',
+  generalError: '',
+  passwordError: ''
+};
+
+const mapDispatchToProps = dispatch => {
+  return {
+    setUser: user => {
+      dispatch(setUser(user))
+    }
+  }
+}
+
+const mapStateToProps = state => {
+  return { user: state.user }
+}
+
+class SignupPage extends Component {
+  static propTypes = {
+    history: PropTypes.object.isRequired,
+    setUser: PropTypes.func.isRequired,
+    user: PropTypes.shape({
+      email: PropTypes.string
```

Comment:
We'll want the entire user object stored in redux. Let's go ahead and check for those properties within the object as well. You can see the entire user object by simply testing the user signup and what we can back on successful signup.